src/native-compile-cache.js

```diff
@@ -38,12 +38,11 @@ class NativeCompileCache {
   }
 
   runInThisContext(code, filename) {
-    // TodoElectronIssue:  produceCachedData is deprecated after Node 10.6, so we'll
-    // will need to update this for Electron v4 to use script.createCachedData().
-    const script = new vm.Script(code, { filename, produceCachedData: true });
+    const script = new vm.Script(code, filename);
+    const cachedData = script.createCachedData();
     return {
       result: script.runInThisContext(),
-      cacheBuffer: script.cachedDataProduced ? script.cachedData : null
+      cacheBuffer: typeof cachedData !== 'undefined' ? cachedData : null
```

Comment:
```suggestion
      cacheBuffer: cachedData !== undefined ? cachedData : null
```

Reply:
You can't directly compare the value of something you expect to be undefined, or you will get this error:
`Uncaught ReferenceError: cachedData is not defined`

`typeof` can check if something is undefined without causing a `ReferenceError`. I don't want to start causing Errors and having to handle them correctly. Best to use `typeof` here.

Reply:
ok will revert it again; thanks for the explanation